src/cbindings.rs

```diff
@@ -0,0 +1,174 @@
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![allow(dead_code)]
+include!(concat!(env!("OUT_DIR"), "/cbindings.rs"));
+
+// Originally based on https://github.com/hexops/mach-dxcompiler/blob/main/src/main.zig
+#[test]
```

Comment:
This is a nice test. Perhaps it should be inside a `#[cfg(test)] mod tests {}` block to be consistent with standard Rust style? There's no need to include this in compilation if tests aren't running, anyhow :)